composer/utils/object_store/uc_object_store.py

```diff
@@ -203,12 +206,27 @@ def get_object_size(self, object_name: str) -> int:
     def list_objects(self, prefix: Optional[str]) -> List[str]:
         """List all objects in the object store with the given prefix.
 
+         .. note::
+
+            This function removes the directories from the returned list.
+
         Args:
             prefix (str): The prefix to search for.
 
         Returns:
             list[str]: A list of object names that match the prefix.
         """
-        # TODO: Implement this function once UC volumes list endpoint is available in the SDK
-        del prefix  # unused
-        raise NotImplementedError(f'{type(self).__name__}.list_objects is not implemented')
+        if not prefix:
```

Comment:
I tried doing that and it's doesn't exactly work because the prefix here is the "bucket" in some sense. It's `Volumes/catalog/schema/volume` and the input could be `prefix="Volumes/catalog/schema/volume/path"` so in order to handle it correctly we would need to use `os.path.commonprefix(..)` which `_get_object_path` is anyways doing.

Reply:
ah ok, I think that makes sense, thanks!